worker/crlWorker/crlWorker.go

```diff
@@ -0,0 +1,117 @@
+package crlWorker
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/mozilla/tls-observatory/certificate"
+	"github.com/mozilla/tls-observatory/worker"
+)
+
+var (
+	workerName = "crlWorker"
+	workerDesc = "Checks certificate CRL (Certificate Revocation List) and reports on certificate revocation"
+)
+
+type Result struct {
+	Revoked bool `json:"revoked"`
+}
+
+type eval struct{}
+
+// Run implements the worker.Worker interface and is called check CRL status
+func (e eval) Run(in worker.Input, resChan chan worker.Result) {
+	result := worker.Result{
+		WorkerName: workerName,
+	}
+	crlRes := Result{}
+
+	// Grab first CRL response we can from our certificate
+	crlResp, err := e.getFirstCRLData(in.Certificate)
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+	certList, err := x509.ParseCRL(crlResp)
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+
+	// Verify our CRL response is signed properly
+	x509Cert, err := in.Certificate.ToX509()
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+	}
+	if err := x509Cert.CheckCRLSignature(certList); err != nil {
```

Comment:
That's the wrong logic, I think. The CRL will be signed by the root or intermediate that issued it, not by the end-entity that is being revoked. The CA doesn't even have access to the private key of the end-entity. If you want to verify the CRL signature, you need to walk the path up from the end-entity to the root(s).